assets/helpers/paymentIntegrations/paymentApi.js

```diff
@@ -0,0 +1,240 @@
+  return { ...getRequestOptions(credentials, csrf), method: 'POST', body: JSON.stringify(data.fields) };
+}
+
+/** Sends a request to the payment API (or support workers, yolo) and converts
+ *  the response into a JSON object */
+function requestPaymentApi(endpoint: string, init: Object) {
+  return logP(fetch(endpoint, init).then(resp => resp.json()));
+}
+
+/** Process the response for a one-off payment from the payment API */
+function checkOneOffStatus(json: Object): Promise<PaymentResult> {
+  if (json.error) {
+    if (json.error.exceptionType === 'CardException') {
+      return Promise.resolve({ paymentStatus: 'failure', error: 'Your card has been declined.' });
+    }
+    const errorHttpCode = json.error.errorCode || 'unknown';
+    const exceptionType = json.error.exceptionType || 'unknown';
+    const errorName = json.error.errorName || 'unknown';
+    return Promise.resolve({
+      paymentStatus: 'failure',
+      error: `Stripe payment attempt failed with following error: code: ${errorHttpCode} type: ${exceptionType} error-name: ${errorName}.`,
+    });
+
+  }
+  return Promise.resolve(PaymentSuccess);
+}
+
+/**
+ * Process the response for a regular payment from the payment API.
+ *
+ * If the payment is:
+ * - pending, then we start polling the API until it's done or some timeout occurs
+ * - failed, then we bubble up an error value
+ * - otherwise, we bubble up a success value
+ */
+function checkRegularStatus(participations: Participations, csrf: CsrfState): Object => Promise<PaymentResult> {
+  const handleCompletion = (json) => {
+    switch (json.status) {
+      case 'success':
+        return PaymentSuccess;
+
+      default:
+        return { paymentStatus: 'failure', error: json.message };
+    }
+  };
+
+  return (json) => {
+    switch (json.status) {
+      case 'pending':
+        return pollUntilP(
+          MAX_POLLS,
+          POLLING_INTERVAL,
+          () => {
+            trackConversion(participations, routes.recurringContribPending);
+            return requestPaymentApi(json.trackingUri, getRequestOptions('same-origin', csrf));
+          },
+          json2 => json2.status === 'pending',
+        ).then(handleCompletion);
+
+      default:
+        return Promise.resolve(handleCompletion(json));
+    }
+  };
+}
+
+/** Returns the URL for one-off payments (should probably be a `const` somewhere) */
+function getOneOffStripeEndpoint() {
+  if (cookie.get('_test_username')) {
+    return addQueryParamsToURL(
+      ONEOFF_CONTRIB_ENDPOINT,
+      { mode: 'test' },
+    );
+  }
+
+  return ONEOFF_CONTRIB_ENDPOINT;
+}
+
+/** Sends a one-off payment request to the payment API and checks the result */
+function postOneOffStripeRequest(data: PaymentFields): Promise<PaymentResult> {
+  return requestPaymentApi(
+    getOneOffStripeEndpoint(),
+    postRequestOptions(data, 'include', null),
+  ).then(checkOneOffStatus);
+}
+
+/** Sends a regular payment request to the payment API and checks the result */
```

Comment:
Currently this request is posted to support-frontend (which provides an API for support-workers). 

_Digression: I think it would make more sense if these endpoints were part of the payment-api, but it didn't exist when we introduced support-workers!_

Reply:
Yeah, I'm using "payment API" glibly to both mean payment API and support-workers 😅

Reply:
I think a new, neutral term would be a lot less confusing!

Reply:
in fact the only place where it's referring to both is in `requestPaymentApi` which as I mentioned above is very general so can be renamed and possibly moved

wherever you're referring to one or the other, for the time being it's best to refer to the specific one (e.g. `recurring contribution endpoint` in this comment)

> Digression: I think it would make more sense if these endpoints were part of the payment-api, but it didn't exist when we introduced support-workers!

Completely agree. Some day!